model/src/main/kotlin/utils/DependencyGraphBuilder.kt

```diff
@@ -179,7 +179,7 @@ class DependencyGraphBuilder<D>(
         )
     }
 
-    private fun Collection<DependencyGraphEdge>.removeCycles(): List<DependencyGraphEdge> {
+    private fun Set<DependencyGraphEdge>.removeCycles(): Set<DependencyGraphEdge> {
```

Comment:
In the spirit of "be lenient when consuming, be strict when producing", should the receiver stay a `Collection`?

Reply:
I believe in general it makes sense, in this case I decided against it so that one does not have to even think about what happens when input contains duplicates, because the API is private (small scope), and because there is no use besides the Sets one.